pywren/storage/storage.py

```diff
@@ -0,0 +1,111 @@
+from __future__ import absolute_import
+import json
+
+from .storage_utils import *
+from .exceptions import *
+from .s3_service import S3Service
+
+
+class Storage(object):
+    """
+    A Storage object is used by executors and other components to access underlying storage service
+    without exposing the the implementation details.
+    Currently we only support S3 as the underlying service.
+    We plan to support other services in the future.
+    """
+
+    def __init__(self, config):
+        self.storage_config = config
+        self.prefix = config['storage_prefix']
+        self.service = config['storage_service']
+        if config['storage_service'] == 's3':
+            self.service_handler = S3Service(config['service_config'])
```

Comment:
StorageBackend sounds better. 
I looked into the python way for interface , and Abstract Base Class, which seems to be the to-go method. The benefit of having an interface class for code/test is probably very limited for a dynamic language, as we probably won't get NotImplementedError from IDE or install time. To that end, this is much worth to add more non-pythonic code?